DOCS-1521: Document Execute Once process behavior (#2381)
viamrobotics/docs: docs/build/configure/_index.md

To automatically run a specified script when the machine boots, configure a _process_.

Start by giving the process a **Name** (`id` in raw JSON) (an identifier of your choice) and clicking **Create process**.

Then fill in the following fields:

<!-- prettier-ignore -->
| Attribute (Builder Mode) | Attribute (Raw JSON) | Type    | Inclusion    | Description |
| ------------------------ | -------------------- | ------- | ------------ | ----------- |
| Executable               | `name`               | string  | **Required** | The command you want to execute when your machine connects to the server. |
| Arguments                | `args`               | string  | Optional     | Arguments to follow the command. |
| Working directory        | `cwd`                | string  | Optional     | Where you want the process to execute. Defaults to the directory where `viam-server` executes. |

> | Logging                  | `log`                | boolean | Optional     | Toggle logging of errors and other messages on or off. |

Reviewer: default missing

> | Logging                  | `log`                | boolean | Optional     | Toggle logging of errors and other messages on or off. |

Reviewer: I assume? 
```suggestion
| Logging                  | `log`                | boolean | Optional     | Toggle logging of errors and other messages on or off. Default: on |
```

Reviewer 2: do we want to just add a column in the table for "default value"?

Author: ```suggestion
| Logging                  | `log`                | boolean | Optional     | Toggle logging of errors and other messages on or off. Default: `true`. |
```

> | Logging                  | `log`                | boolean | Optional     | Toggle logging of errors and other messages on or off. Default: `true`. |

Author: ```suggestion
| Logging                  | `log`                | boolean | Optional     | Toggle logging of errors and other messages on or off. Default: `false`. |
```

> | Execute once             | `one_shot`           | boolean | Optional     | Toggle whether to execute the command just once or keep running the process indefinitely.<ul><li>If true, the process executes once at `viam-server` startup. Until the process completes, `viam-server` startup is blocked and the robot appears offline in the [Viam app](https://app.viam.com), so this should only be used for quick processes.</li><li>If false, the process continues to run. If the process crashes, it automatically restarts. It does not block `viam-server`. False is the default.</li></ul> |

Reviewer: I think that's how we often do it?
```suggestion
| Execute once             | `one_shot`           | boolean | Optional     | Toggle whether to execute the command just once or keep running the process indefinitely.<ul><li>If true, the process executes once at `viam-server` startup. Until the process completes, `viam-server` startup is blocked and the robot appears offline in the [Viam app](https://app.viam.com), so this should only be used for quick processes.</li><li>If false, the process continues to run. If the process crashes, it automatically restarts. It does not block `viam-server`. Default: `false`.</li></ul> |
```

To automatically run a specified script when the machine boots, configure a _process_.

Start by giving the process a **Name** (`id` in raw JSON) (an identifier of your choice) and clicking **Create process**.

Then fill in the following fields:

- **Executable** (`name`): The command you want to execute when your machine connects to the server.
- **Arguments** (`args`): Optional arguments to follow the command.
- **Working directory** (`cwd`): Where you want the process to execute.
  An optional setting that defaults to the directory where `viam-server` executes.

> - **Logging** (`log`): Toggle logging of errors and other messages on or off (boolean).

Reviewer: I think the (`boolean`) at the end is odd and not consistent with the other list items which dont have `(string)`. Would a table be more suitable and consistent with how we generally document config fields?
```suggestion
- **Logging** (`log`): Toggle logging of errors and other messages on or off.
```

Author: I can try a table and see how it looks. I think this was originally written with a more how-to style in mind, sort of consistent with how we write tutorials with unordered lists, or how https://docs.viam.com/data/capture/#add-the-data-management-service uses steps and sentences. This one is also weird in that the builder labels don't match the raw JSON at all.

> - **Execute once** (`one_shot`): Toggle whether to execute the command just once or keep running the process indefinitely (boolean).

Reviewer: Did you speak to eng already on this? I thought it was more about restarting it in case it stops rather than running indefinitely.
```suggestion
- **Execute once** (`one_shot`): Toggle whether to execute the command just once or keep restarting the process indefinitely.
```

Author: I was under the impression that it's about whether it runs and completes before RDK starts up, versus whether it continues to run in parallel with RDK.

>   - If true, the process executes once at `viam-server` startup.

Reviewer: ```suggestion
  - If `true`, the process executes once at `viam-server` startup.
```

Author: Thought leaving as not-monospace is more inclusive of the fact that most or many users will use builder mode, which has no such `true` string?

    Until the process completes, `viam-server` startup is blocked and the robot appears offline in the [Viam app](https://app.viam.com), so this should only be used for quick processes.

>   - If false, the process continues to run.

Reviewer: did you get this info from eng or do you know this?

Author: Yeah--see the ticket comment, as well as the custom robot dog tutorial
